routing/router.go

```diff
@@ -1849,6 +1849,11 @@ func (r *ChannelRouter) sendPayment(payment *LightningPayment,
 			case *lnwire.FailUnknownPaymentHash:
 				return preImage, nil, sendError
 
+			// If we sent the wrong amount to the destination, then
+			// we'll exit early.
+			case *lnwire.FailIncorrectPaymentAmount:
+				return preImage, nil, sendError
+
```

Comment:
should this be squashed with the prior commit?

Reply:
Yeah I'll squash in all the fixups into the original commit that starts the switch over.